spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/source/TestStructuredStreamingRead3.java

```diff
@@ -247,6 +249,46 @@ public void testReadingStreamFromTimestampStartWithExistingTimestamp() throws Ex
     Assertions.assertThat(actual).containsExactlyInAnyOrderElementsOf(Iterables.concat(expected));
   }
 
+  @SuppressWarnings("unchecked")
+  @Test
+  @Ignore("Existing logic doesn't work when the expired snapshot gets deleted")
```

Comment:
Shouldn't we add a test for checking the new functionality? I would rather we didn't just add in a test that we don't run.

As I mentioned above you may want to add in a test for reading a stream from a timestamp after a rollback, so there exists old snapshots which are not live.

Reply:
Hello @RussellSpitzer ,

Thanks for reviewing. This test tries to reproduce the problem @rdblue mentioned above, while some snapshot expired, the the "parent" will break in some point with NPE since the snapshot is not available anymore. And I don't find a good way to address this problem yet, so created the unit test for reproducible purpose. The PR is still in progressing, once the logic is settled down this test will either remove or removing the Ignore tag

Reply:
Sounds good to me, ping me again when you want me to take another look :)

Reply:
@RussellSpitzer Thanks a lot
I just push another PR to address the NPE issue and move the logic back to parent looks up.
Please take a look when you have time